third-party/libscan/libscan/ebook/ebook.c

```diff
@@ -225,34 +223,19 @@ static int read_stext_block(fz_stext_block *block, text_buffer_t *tex) {
     return 0;
 }
 
-#define IS_VALID_BPP(d) ((d)==1 || (d)==2 || (d)==4 || (d)==8 || (d)==16 || (d)==24 || (d)==32)
+static void fill_image_ocr_cb(const char* text, size_t len) {
+  text_buffer_append_string(&thread_buffer, text, len - 1);
+}
 
 void fill_image(fz_context *fzctx, UNUSED(fz_device *dev),
                 fz_image *img, UNUSED(fz_matrix ctm), UNUSED(float alpha),
                 UNUSED(fz_color_params color_params)) {
 
     int l2factor = 0;
 
-    if (img->w > MIN_OCR_SIZE && img->h > MIN_OCR_SIZE && IS_VALID_BPP(img->n)) {
-
+    if (img->w > MIN_OCR_SIZE && img->h > MIN_OCR_SIZE && OCR_IS_VALID_BPP(img->n)) {
         fz_pixmap *pix = img->get_pixmap(fzctx, img, NULL, img->w, img->h, &l2factor);
-
-        if (pix->h > MIN_OCR_SIZE && img->h > MIN_OCR_SIZE && img->xres != 0) {
```

Comment:
typo here? img->w, img->h in outer loop, so pix->w, pix->h here?

Reply:
Good catch, you're right it should be `if (pix->w > MIN_OCR_SIZE && pix->h > MIN_OCR_SIZE && img->xres != 0) {`